src/esp/assets/ResourceManager.cpp

```diff
@@ -822,109 +872,149 @@ bool ResourceManager::loadPTexMeshData(const AssetInfo& info,
 #endif
 }
 
-// semantic instance mesh import
-bool ResourceManager::loadInstanceMeshData(
-    const AssetInfo& info,
+scene::SceneNode* ResourceManager::createRenderAssetInstancePTex(
+    const RenderAssetInstanceCreation& creation,
     scene::SceneNode* parent,
-    DrawableGroup* drawables,
-    bool computeAbsoluteAABBs,
-    bool splitSemanticMesh /* = true */) {
-  if (info.type != AssetType::INSTANCE_MESH) {
-    LOG(ERROR) << "loadInstanceMeshData only works with INSTANCE_MESH type!";
-    return false;
+    DrawableGroup* drawables) {
+  ASSERT(!creation.scale);  // PTex doesn't support scale
+  ASSERT(creation.lightSetupKey ==
+         NO_LIGHT_KEY);  // PTex doesn't support lighting
+
+  const std::string& filename = creation.filepath;
+  const LoadedAssetData& loadedAssetData = resourceDict_.at(creation.filepath);
+  const MeshMetaData& metaData = getMeshMetaData(filename);
+  const auto& info = loadedAssetData.assetInfo;
+  auto indexPair = metaData.meshIndex;
+  int start = indexPair.first;
+  int end = indexPair.second;
+  std::vector<StaticDrawableInfo> staticDrawableInfo;
+
+  scene::SceneNode* instanceRoot = &parent->createChild();
+
+  for (int iMesh = start; iMesh <= end; ++iMesh) {
+    auto* pTexMeshData = dynamic_cast<PTexMeshData*>(meshes_[iMesh].get());
+
+    pTexMeshData->uploadBuffersToGPU(false);
+
+    for (int jSubmesh = 0; jSubmesh < pTexMeshData->getSize(); ++jSubmesh) {
+      scene::SceneNode& node = instanceRoot->createChild();
```

Comment:
This is one of the few behavior changes in the PR. For the instance mesh, instead of adding all these mesh child sceneNodes directly to the parent sceneNode, I first create an `instanceRoot` sceneNode. I want every render asset instance to have a single root sceneNode. This is needed for render replay.

I acknowledge there's probably a slight perf cost. Long-term, our new Vulkan batched renderer probably won't use a SceneNode hierarchy, so this shouldn't matter.